pkg/plugin/v3/scaffolds/internal/templates/config/manager/config.go

```diff
@@ -38,11 +42,20 @@ func (f *Config) SetTemplateDefaults() error {
 		f.Path = filepath.Join("config", "manager", "manager.yaml")
 	}
 
-	f.TemplateBody = configTemplate
+	args := argsFragment
+	if f.ComponentConfig {
+		args = ""
+	}
+
+	f.TemplateBody = fmt.Sprintf(configTemplate, args)
```

Comment:
Same as in `main`, template inside the template body with `{{ if .ComponentConfig }}` instead of in `SetTemplateDefaults`.